examples/llm/aiq_baseline/langchain_corpus_ingest.py

```diff
@@ -0,0 +1,260 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+from glob import glob
+
+import torch
+from langchain_text_splitters import RecursiveCharacterTextSplitter
+from openai import OpenAI
+from pymilvus import MilvusClient
+from tqdm import tqdm
+
+from torch_geometric.nn import LLMJudge, SentenceTransformer
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+)
+logger = logging.getLogger(__name__)
+
+# Define constants for better readability
+DEFAULT_ENDPOINT_URL = "https://integrate.api.nvidia.com/v1"
+ENCODER_MODEL_NAME_DEFAULT = "Alibaba-NLP/gte-modernbert-base"
+LLM_GENERATOR_NAME_DEFAULT = "nvidia/llama-3.1-nemotron-70b-instruct"
+NV_NIM_MODEL_DEFAULT = "nvidia/llama-3.1-nemotron-ultra-253b-v1"
+
+
+def test(milvus_client, encoder_model, metric_type: str, dataset: str,
+         collection_name: str, llm_generator_name: str):
+
+    with open(os.path.join(dataset, "train.json")) as file:
+        qa_pairs = json.load(file)
+
+    score = []
+    for data_point in tqdm(qa_pairs, desc="Retrieving pairs"):
+        if data_point["is_impossible"]:
+            continue
+        question, answer = (data_point["question"], data_point["answer"])
+
+        print("The question = \n", question)
+        search_res = milvus_client.search(
+            collection_name=collection_name,
+            data=[
+                # Convert the question to an embedding vector
+                emb_text(question, encoder_model).tolist()[0]
+            ],  # Use the `emb_text` function to convert the question to an embedding vector
+            limit=10,  # Return top 3 results
```

Comment:
Is this right?
```suggestion
            limit=3,  # Return top 3 results
```
or should we do this?
```suggestion
            limit=10,  # Return top 10 results
```